esmvaltool/diag_scripts/seaborn_diag.py

```diff
@@ -174,9 +174,36 @@ def _create_plot(
         plt.suptitle(cfg["suptitle"], y=1.05)
     if cfg["legend_title"] is not None:
         _set_legend_title(plot_obj, cfg["legend_title"])
+    if plot_func_str == "jointplot" and plot_kwargs["cbar"]:
+        # Reposition colorbar so it is to the right of marginals plot
+        plt.subplots_adjust(left=0.1, right=0.8, top=0.9, bottom=0.1)
+        # get the current positions of the joint ax and the ax for
+        # the marginal x
+        pos_joint_ax = plot_obj.ax_joint.get_position()
+        pos_marg_x_ax = plot_obj.ax_marg_x.get_position()
+        # reposition the joint ax so it has the same width as the
+        # marginal x ax
+        plot_obj.ax_joint.set_position(
+            [
+                pos_joint_ax.x0,
+                pos_joint_ax.y0,
+                pos_marg_x_ax.width,
+                pos_joint_ax.height,
+            ]
+        )
+        # reposition the colorbar using new x positions and y
+        # positions of the joint ax
+        plot_obj.fig.axes[-1].set_position(
+            [0.83, pos_joint_ax.y0, 0.07, pos_joint_ax.height]
+        )
 
     # Save plot
-    plot_path = get_plot_filename(f"seaborn_{plot_func_str}", cfg)
+    if cfg["plot_filename"] is not None:
+        plot_path = get_plot_filename(
+            f"seaborn_{plot_func_str}_{cfg['plot_filename']}", cfg
+        )
+    else:
+        plot_path = get_plot_filename(f"seaborn_{plot_func_str}", cfg)
```

Comment:
```suggestion
    plot_path = get_plot_filename(cfg["plot_filename"], cfg)
```

You also need to make sure that this option always exists (regardless of the recipe), so please add the option `plot_filename` with the default `f"seaborn_{cfg.get('seaborn_func', '')}"` to the function `_get_default_cfg` in this diagnostic.

Reply:
Thanks @schlunma for the suggestion! I changed it accordingly.

Reply:
Thanks Lisa! Could you also please add the option to the beginning of the file where all possible diagnostic options are documented?

Reply:
Sorry @schlunma, done.

Reply:
Thanks!! No worries ☺️